File: cs/cs-kata/Bowling180703.cs
using System;
using System.Collections.Generic;

namespace cs_kata
{
    public class Bowling180703
    {
        private List<int> rolls = new List<int>();

        public void Roll(int pins)
        {
            this.rolls.Add(pins);
        }

        public int Score()
        {
            int score = 0;
            int rollIndex = 0;
            for (int frame = 0; frame < 10; frame++) {
                if (this.isStrike(rollIndex)) {
                    score += this.scoreStrike(rollIndex);
                    rollIndex += 1;
                } else if (this.isSpare(rollIndex)) {
                    score += this.scoreSpare(rollIndex);
                    rollIndex += 2;
                } else {
                    score += this.scoreNormalFrame(rollIndex);
                    rollIndex += 2;
                }
            }
            return score;
        }

        private bool isStrike(int rollIndex)
        {
            return 10 == this.rolls[rollIndex];
        }

        private bool isSpare(int rollIndex)
        {
            return 10 == (this.rolls[rollIndex] + this.rolls[rollIndex + 1]);
        }

        private int scoreStrike(int rollIndex)
        {
            return 10 + this.rolls[rollIndex + 1] + this.rolls[rollIndex + 2];
        }

        private int scoreSpare(int rollIndex)
        {
            return 10 + this.rolls[rollIndex + 2];
        }

        private int scoreNormalFrame(int rollIndex) 
        {
            return this.rolls[rollIndex] + this.rolls[rollIndex + 1];
        }
    }
}

File: swift/CodeKataPracticeSwift/CodeKataPracticeSwift/Bowling180703.swift
//
//  Bowling180703.swift
//  CodeKataPracticeSwift
//
//  Created by toto on 2018/07/03.
//  Copyright © 2018 toto. All rights reserved.
//

import Foundation

class Bowling180703 {

    var rolls = [Int]()

    func roll(pins: Int) {
        rolls.append(pins)
    }

    func score() -> Int {
        var score = 0
        var rollIndex = 0
        for _ in 0..<10 {
            if isStrike(rollIndex: rollIndex) {
                score += scoreStrike(rollIndex: rollIndex)
                rollIndex += 1
            } else if isSpare(rollIndex: rollIndex) {
                score += scoreSpare(rollIndex: rollIndex)
                rollIndex += 2
            } else {
                score += scoreNormalFrame(rollIndex: rollIndex)
                rollIndex += 2
            }
        }
        return score;
    }

    func isStrike(rollIndex: Int) -> Bool {
        return 10 == rolls[rollIndex]
    }

    func isSpare(rollIndex: Int) -> Bool {
        return 10 == (rolls[rollIndex] + rolls[rollIndex + 1])
    }

    func scoreStrike(rollIndex :Int) -> Int {
        return 10 + rolls[rollIndex + 1] + rolls[rollIndex + 2]
    }

    func scoreSpare(rollIndex: Int) -> Int {
        return 10 + rolls[rollIndex + 2]
    }

    func scoreNormalFrame(rollIndex: Int) -> Int {
        return rolls[rollIndex] + rolls[rollIndex + 1]
    }
}
